rvgo/fast/memory.go

```diff
@@ -200,21 +199,6 @@ func (m *Memory) SetUnaligned(addr uint64, dat []byte) {
 	if d == len(dat) {
 		return // if all the data fitted in the page, we're done
 	}
-
-	// continue to remaining part
```

Comment:
If we're removing the ability to do unaligned writes, can we also add a check at the top of `SetAligned` that the write is 8-byte aligned?

I see that in the VM, we check for 4-byte alignment, but it is possible to read over multiple pages if we're reading full double words and only enforcing 4-byte alignment.

Reply:
I ran some riscv tests and observed the memory write patterns. 
- the write operations aren't necessarily 8-byte (or even 4-byte) aligned. 
- however, the the write themselves do not flow over the page boundaries as far as I've tested. 
- So it seems maybe the ability to write over the page boundary isn't really needed.

However, since the addresses aren't really aligned and there may exist some cases which the writes could flow over the page boundary, I think it'd be most appropriate to return the previous behavior of SetUnaligned. What do you think?

Reply:
I agree with returning to the previous behavior. It's a bit more in-line with the ISA spec, and places less assumptions on the compilers.